### ISSUE TITLE
[NhkVod] [Pixsel] Video appears to download but doesn't play

### ISSUE THREAD

**Issue by mmanjos** [yt-dlp/yt-dlp#3666]
### Checklist

- [X] I'm reporting a broken site
- [X] I've verified that I'm running yt-dlp version **2022.04.08** ([update instructions](https://github.com/yt-dlp/yt-dlp#update)) or later (specify commit)
- [X] I've checked that all provided URLs are alive and playable in a browser
- [X] I've checked that all URLs and arguments with special characters are [properly quoted or escaped](https://github.com/ytdl-org/youtube-dl#video-url-contains-an-ampersand-and-im-getting-some-strange-output-1-2839-or-v-is-not-recognized-as-an-internal-or-external-command)
- [X] I've searched the [bugtracker](https://github.com/yt-dlp/yt-dlp/issues?q=) for similar issues including closed ones. DO NOT post duplicates
- [X] I've read the [guidelines for opening an issue](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#opening-an-issue)
- [X] I've read about [sharing account credentials](https://github.com/yt-dlp/yt-dlp/blob/master/CONTRIBUTING.md#are-you-willing-to-share-account-details-if-needed) and I'm willing to share it if required

### Region

US

### Description

tested on Linux (env in log) with commit 5aca3d73f64649f4ff460a791a7e20b9aca616cb as well as 2022.04.08

The video file appears to download fully, but it is unwatchable - it can't be read by mpv, ffmpeg, handbrake, vlc, etc

The video is watchable in my browser

Download URL is https://www3.nhk.or.jp/nhkworld/en/ondemand/video/2054145/

When I play the 714.38MiB file (which I always receive as sha256 hash 983220876ef8f8fab6901bd297c9c41d2b6eca128251b7dbb8365d4c4b6a555e - I get the same downloaded file hash from multiple computers), I get the following output from mpv:

$ mpv Trails\ to\ Oishii\ Tokyo\ -\ SHIRASU\ \[f11025m6\].mp4 
[ffmpeg/demuxer] mov,mp4,m4a,3gp,3g2,mj2: moov atom not found
[lavf] avformat_open_input() failed
[ffmpeg/demuxer] mov,mp4,m4a,3gp,3g2,mj2: moov atom not found
[lavf] avformat_open_input() failed
Failed to recognize file format.

Exiting... (Errors when loading file)

### Verbose log

```shell
[debug] Command-line config: ['-vU', 'https://www3.nhk.or.jp/nhkworld/en/ondemand/video/2054145/']
[debug] Encodings: locale UTF-8, fs utf-8, out utf-8, err utf-8, pref UTF-8
[debug] yt-dlp version 2022.04.08 [7884ade65]
[debug] Lazy loading extractors is disabled
[debug] Python version 3.10.4 (CPython 64bit) - Linux-5.15.36-1-lts-x86_64-with-glibc2.35
[debug] Checking exe version: ffprobe -bsfs
[debug] Checking exe version: ffmpeg -bsfs
[debug] exe versions: ffmpeg 5.0 (setts), ffprobe 5.0, rtmpdump 2.4
[debug] Optional libraries: Cryptodome-3.12.0, certifi-2021.10.08, sqlite3-2.6.0
[debug] Proxy map: {}
Latest version: 2022.04.08, Current version: 2022.04.08
yt-dlp is up to date (2022.04.08)
[debug] [NhkVod] Extracting URL: https://www3.nhk.or.jp/nhkworld/en/ondemand/video/2054145/
[NhkVod] 2054-145: Downloading JSON metadata
[debug] [Piksel] Extracting URL: https://player.piksel.com/v/refid/nhkworld/prefid/nw_vod_v_en_2054_145_20220504233000_01_1651676711
[Piksel] nw_vod_v_en_2054_145_20220504233000_01_1651676711: Downloading webpage
[Piksel] nw_vod_v_en_2054_145_20220504233000_01_1651676711: Downloading JSON metadata
[Piksel] nw_vod_v_en_2054_145_20220504233000_01_1651676711: Downloading JSON metadata
[Piksel] f11025m6: Downloading m3u8 information
[Piksel] f11025m6: Downloading SMIL file
[debug] Sort order given by extractor: tbr
[debug] Formats sorted by: hasvid, ie_pref, tbr, lang, quality, res, fps, hdr:12(7), vcodec:vp9.2(10), acodec, filesize, fs_approx, vbr, abr, asr, proto, vext, aext, hasaud, source, id
[debug] Default format spec: bestvideo*+bestaudio/best
[info] f11025m6: Downloading 1 format(s): http-10268
[debug] Invoking downloader on "https://mz-nhk10.mzpf.webcdn.stream.ne.jp/www60/mz-nhk10/media/mm/flvmedia/5905/m/F/m/mF_mp4file10508399-3973200.mp4?cid=5905&aid=3973200&afid=8383004&assetid=3973200"
[download] Destination: Trails to Oishii Tokyo - SHIRASU [f11025m6].mp4
[download] 100% of 714.38MiB in 02:19


$ mpv Trails\ to\ Oishii\ Tokyo\ -\ SHIRASU\ \[f11025m6\].mp4 
[ffmpeg/demuxer] mov,mp4,m4a,3gp,3g2,mj2: moov atom not found
[lavf] avformat_open_input() failed
[ffmpeg/demuxer] mov,mp4,m4a,3gp,3g2,mj2: moov atom not found
[lavf] avformat_open_input() failed
Failed to recognize file format.

Exiting... (Errors when loading file)
```

**Comment by mmanjos** [comment]
There's definitely some good video data in the downloaded file - I was able to partially recover it using `untrunc` but I don't have enough mp4/container editing skills to fully determine what's going on here.

**Comment by twpol** [comment]
I've seen similar from other NHK videos, the key thing I noticed was the file size downloaded (714 MiB in your case) is smaller than the size listed next to the format (`http-10268`) in `--list-formats`, often quite significantly

I have always had to manually pick a lower-quality format to get a working video; it feels like their CDN has a truncated file or something, but I've never looked inside the file to see if that's really the case

**Comment by Lesmiscore** [comment]
Did you attempted to use a different formats, like HLS?

**Comment by mmanjos** [comment]
Thanks @twpol and @Lesmiscore - the lower quality formats definitely do work, so I agree it's probably an issue with their CDN that their web player probably heals over or handles differently. 

For anyone else experiencing this issue, NHK downloads seem to work whenever I set `-f [width<=1280]` (multiple videos there are posted with different stream formats, so matching "best less than 1280px width" seems to work best if mirroring their library.

Closing for now - I don't think this is a bug with yt-dlp. Thanks again!
